File: src/test/resources/kt/InlineMathOperations.kt
class InlineMathOperations {
    private fun complexCalculation(a: Int, b: Int): Int {
        val sum = a + b
        val product = a * b
        val difference = a - b
        return sum + product - difference
    }

    fun calculateResult1(x: Int, y: Int): Int {
        return complexCalculation(x, y) + 10
    }

    fun calculateResult2(p: Int, q: Int): Int {
        return 2 * complexCalculation(p, q)
    }
}

File: src/test/resources/go/InlineMathOperations.go
package main

import "fmt"

type InlineMathOperations struct{}

func (self InlineMathOperations) complexCalculation(a, b int) int {
	sum := a + b
	product := a * b
	difference := a - b
	return (sum + product - difference)
}

func (self InlineMathOperations) calculateResult1(x, y int) int {
	return self.complexCalculation(x, y) + 10
}

func (self InlineMathOperations) calculateResult2(p, q int) int {
	return 2 * self.complexCalculation(p, q)
}
